$NetBSD: patch-ae,v 1.2 2007/03/17 14:37:47 rillig Exp $

gcc4 complains:
kopt.c:34: error: static declaration of 'krbONE' follows non-static declaration
.../.buildlink/include/kerberosIV/krb.h:64: error: previous declaration of 'krbONE' was here

--- server/kopt.c.orig	2001-02-27 05:48:01.000000000 +0100
+++ server/kopt.c	2007-03-17 15:34:42.000000000 +0100
@@ -31,8 +31,8 @@ static const char *rcsid_rd_req_c =
 
 /* Byte ordering */
 #undef HOST_BYTE_ORDER
-static int krbONE = 1;
-#define		HOST_BYTE_ORDER	(* (char *) &krbONE)
+static int my_krbONE = 1;
+#define		HOST_BYTE_ORDER	(* (char *) &my_krbONE)
 
 #define		KRB_PROT_VERSION 	4
 
@@ -161,7 +161,7 @@ void add_to_key_sched_cache(key, sched)
 
 int
 krb_set_key(key,cvt)
-    char *key;
+    void *key;
     int cvt;
 {
 #ifdef NOENCRYPTION
@@ -172,7 +172,7 @@ krb_set_key(key,cvt)
     int ret;
 
     if (cvt)
-	string_to_key(key,serv_key);
+	string_to_key((char *)key,serv_key);
     else
 	memcpy((char *)serv_key,key,8);
 
@@ -187,6 +187,14 @@ krb_set_key(key,cvt)
 #endif /* NOENCRYPTION */
 }
 
+static u_long
+swap_u_long(u_long x)
+{
+	return ((x & 0xff000000) >> 24) |
+	    ((x & 0x00ff0000) >> 8) |
+	    ((x & 0x0000ff00) << 8) |
+	    ((x & 0x000000ff) << 24);
+}
 
 /*
  * krb_rd_req() takes an AUTH_MSG_APPL_REQUEST or
@@ -234,7 +242,7 @@ krb_rd_req(authent,service,instance,from
     KTEXT authent;			/* The received message */
     char *service;			/* Service name */
     char *instance;			/* Service instance */
-    unsigned KRB_INT32 from_addr;	/* Net address of originating host */
+    KRB_INT32 from_addr;		/* Net address of originating host */
     AUTH_DAT *ad;			/* Structure to be filled in */
     char *fn;				/* Filename to get keys from */
 {
